activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/main/java/org/activiti/cloud/services/identity/keycloak/client/KeycloakClient.java

```diff
@@ -238,4 +240,22 @@ void updateClient(
         @PathVariable("id") String id,
         @RequestBody KeycloakClientRepresentation keycloakClientRepresentation
     );
+
+    @RequestMapping(method = RequestMethod.POST, value = "/clients/{id}/client-secret")
+    @Headers("Content-Type: application/json")
+    KeycloakCredentialRepresentation createClientSecretById(
+        @RequestBody KeycloakCredentialRequestRepresentation requestRepresentation,
+        @PathVariable("id") String id
+    );
+
+    @RequestMapping(method = RequestMethod.GET, value = "/clients/{id}/client-secret")
+    @Headers("Content-Type: application/json")
+    KeycloakCredentialRepresentation getClientSecretById(@PathVariable("id") String id);
```

Comment:
Is this method needed? It looks strange to me that we publish an endpoint where you can retrieve the client secret

Reply:
OK I see it exists in the Keycloak API